src/main/java/org/apache/maven/plugins/dependency/AbstractDependencyMojo.java

```diff
@@ -59,6 +60,21 @@
     @Component
     private ArchiverManager archiverManager;
 
+
+    /**
+     * For m2e incremental build support
```

Comment:
Since the `BuildContext` API is not Eclipse M2E specific (even though M2E is the only user I'm aware of) I suggest to make this comment more generic.
The same applies for the comment of the `skipDuringIncrementalBuild`.

Reply:
It is mostly to not confuse it with http://takari.io/2014/03/25/incremental-build.html. Any suggestion for the naming as just `incremental build support` may be a bit too generic (as there are other incremental build APIs)?

Reply:
Maybe `IDE build support` as this is also not only about incremental support but also about other features from the Build Support API (like notification about updates files or context-specific error messages)

Reply:
Sounds good.